### DISCUSSION TITLE
replaceall regex does not produce result for control characters in trino CLI

### DISCUSSION THREAD

**Discussion by natarajank11** [trinodb/trino#17673] [General]
Trino UDF (version 407):
Basically i want to remove control characters from the string.

I tried following regex pattern,
`String regex = "[\\\\p{C}]";`

Trino result,
 `select rm_code('hello\bhello\thello'); => hellobhellothello`

This ticket is similar to https://github.com/trinodb/trino/discussions/17474

**Comment by hashhar** [comment]
SQL varchar literals don't have control characters so your regex will never match.

The `\b` is treated literally as `\b` -> `SELECT 'hello\bworld'; -- hello\bworld`

So you need to change your regex to: `String regex = "\\\\b";` to match the `\b` in varchar literals.

BTW since SQL varchar literals don't have any control characters why do you want to "remove control characters" and how do you plan to differentiate between `hello\world` and `hello\wworld` (i.e. a case where the `\` is a literal slash and not part of an escape).
